### PR TITLE
add support for csky arch(elf)

### PR DESCRIPTION
**Pull request by Dirreke**

### CODE REVIEW

**Comment by bjorn3** [comment]
All matches seem to be sorted by variant name. Could you insert the Csky variant at the right place to keep it sorted?

File: src/common.rs [hunk 1]
```diff
@@ -29,6 +29,7 @@ pub enum Architecture {
     Wasm32,
     Wasm64,
     Xtensa,
+    CSKY,
```
Comment (bjorn3) [inline src/common.rs:32]:
```suggestion
    Csky,
```
Rustc warns about non-CamelCase names.

**Review by philipc — CHANGES_REQUESTED** [review]
Thanks for the PR. Please also add support in these two places:
https://github.com/gimli-rs/object/blob/b237c5c603bda4fc77eec4db951d833b79cb543d/src/read/elf/file.rs#L154
https://github.com/gimli-rs/object/blob/b237c5c603bda4fc77eec4db951d833b79cb543d/tests/round_trip/mod.rs#L231

**Review by philipc — APPROVED** [review]

**Comment by Dirreke** [comment]
When will the next version be released

**Comment by philipc** [comment]
I don't think there's been any breaking changes yet, so I can do a point release when needed.

**Comment by philipc** [comment]
0.32.0 has been published.

**Pull request merged**